config.go

```diff
@@ -154,6 +158,16 @@ type ClientConfig struct {
 
 	DisableWebtorrent bool
 	DisableWebseeds   bool
+	// HTTPClient is used to override default client which is used for
+	// making requests to webseeds and torrent file downloads. This helps
+	// to provide custom logic like metrics, tracing collection, using custom
+	// transports, rate limiting policies, semaphores, etc.
+	// Probably this has to be used to request webseeds only. A client which
+	// requests a torrent files can have different requirements (need an advice here).
+	HTTPClient webseed.HttpClient
```

Comment:
There was a HttpClient exposed a long while ago, and it was broken instead into HTTPProxy and HTTPUserAgent. Do I read this correctly in that if a http Client is provided, it supersedes all those other options? It also looks like it will be used for trackers and sources too?